src/components/views/dialogs/InviteDialog.tsx

```diff
@@ -76,18 +83,43 @@ class Member {
      * avatar MXC URL or null if none set. For 3PIDs this should always be null.
      */
     getMxcAvatarUrl(): string { throw new Error("Member class not implemented"); }
+    /**
+    * Determines whether or not member listed is a favorited member
+    * Unless value is provided, default is set to false
+    */
+    get favorite(): boolean { throw new Error("Member class not implemented"); }   // determines whether or not member is favorite
+
+    /**
+     * Determines whether or not given member is available for chat
+     * Unless value is provided default is set to false
+    */
+    get available(): boolean { throw new Error("Member class not implemented"); }  // determines whether or not to initiate chats
+
+    /**
+     * Determines what is category of current role category
+     * It is used by rendering right avatar color according to
+     * category id, also used by name initials
+     * Default will be set to null
+     */
+    get roleCategoryId(): string { throw new Error("Member class not implemented"); }
 }
 
 class DirectoryMember extends Member {
     _userId: string;
     _displayName: string;
     _avatarUrl: string;
+    _isFavorite: boolean;
+    _isAvailable: boolean;
+    _roleCategoryId: string;
 
-    constructor(userDirResult: {user_id: string, display_name: string, avatar_url: string}) {
+    constructor(userDirResult: {user_id: string, display_name: string, avatar_url: string, favorite: boolean, available: boolean, roleCategoryId: string}) {
```

Comment:
Can we overload the constructor, so the defaults of false, false, null can be centralised?

Reply:
The value has been set to false in constructor and made optional for string as well and also put my changes in new line which I should have done earlier. Using question mark '?' will make it optional and earlier befor eyou had pointed it out, I had tested with matrix default dialog window where invit is sent and it works well so where objec tmember is called, null default is set for string and false for booleans. But with your recommendation I made one more final change to make these values optional in constructor and tested it out again. Works well again with matrix default send invite to user window which opens from left plus menu. I have continued to use this window until i finish directory work so that I can test if default behavior is still kept same by making my implementations optional fields. Based on  my bunch of previous refactors it has shown that using question mark is making values optional wherever it is used.

Reply:
Should roleCategoryId also have the question mark?